site/app/controllers/student/TeamController.php

```diff
@@ -6,9 +6,21 @@
 use app\controllers\admin\AdminGradeableController;
 use app\libraries\FileUtils;
 use app\libraries\response\RedirectResponse;
+use app\models\Team;
 use Symfony\Component\Routing\Annotation\Route;
 
 class TeamController extends AbstractController {
+    /**
+     * Check if a string has special characters.
+     * @param string $str
+     * @return bool
+     */
+    public static function hasSpecialCharacters($str) {
+        $allowedCharacters = ['a-z', 'A-Z', '0-9', '-', '_', ' ']; // This list can be expandable upon further requests
+        $pattern = '/[^' . implode('', $allowedCharacters) . ']/';
```

Comment:
Why do you need an array of allowed characters here?

Reply:
@MasterOdin suggested to maintain a list of allowed characters 
such that it would be easy to update it in the future rather than changing the regex itself.

Reply:
Sorry, when I said "list" I meant it more abstractly, rather than specifically that you should use a list/array to have the characters. I think it's fine to just have this singular regex, as editing the range is easy enough.

```suggestion
        $pattern = '/[^a-zA-Z0-9-_ ]/';
```

Reply:
> Sorry, when I said "list" I meant it more abstractly, rather than specifically that you should use a list/array to have the characters. I think it's fine to just have this singular regex, as editing the range is easy enough.

got it
pushed the changes
and you may review it